SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetricsDisplay.cpp

```diff
@@ -165,16 +165,15 @@ void ASpatialMetricsDisplay::Tick(float DeltaSeconds)
 
 		for (const FWorkerStats& OneWorkerStats : WorkerStats)
 		{
-			const float TimeSinceUpdate = CurrentTime - WorkerStatsLastUpdateTime[OneWorkerStats.WorkerName];
-
-			if (TimeSinceUpdate > DropStatsIfNoUpdateForTime)
+			if (ShouldRemoveStats(SpatialNetDriver->Time, OneWorkerStats))
 			{
 				WorkerStatsToRemove.Add(OneWorkerStats);
```

Comment:
I understand it's not directly related to this PR, however I'm curious what is a benefit of delaying the WorkerStats removal (by adding them to WorkerStatsToRemove and deleting later) instead of doing them directly here?

Reply:
it's to avoid any issues related to modifying the collection while iterating over it.  Alternatively could iterate over the array backwards